app/src/main/java/cm/aptoide/pt/ads/IronSourceAdRepository.java

```diff
@@ -0,0 +1,87 @@
+package cm.aptoide.pt.ads;
+
+import android.app.Activity;
+import android.util.Log;
+import cm.aptoide.pt.BuildConfig;
+import com.ironsource.mediationsdk.IronSource;
+import com.ironsource.mediationsdk.logger.IronSourceError;
+import com.ironsource.mediationsdk.sdk.InterstitialListener;
+import rx.subjects.PublishSubject;
+
+public class IronSourceAdRepository {
+  private static final String TAG = "IronSourceAdRepository";
+  private final Activity activity;
+  private final PublishSubject<Void> impressionSubject;
+  private final PublishSubject<Void> clickSubject;
```

Comment:
Should avoid multiple subjects - we could use only 1 with an Event object that would have information about whether it is an impression or a click